**Задача 1.**

Имеется колода в 52 карты. Найти число возможностей вытянуть из неё 4 карты так, чтобы среди них был хотя бы один туз.

Решение:
    Не менее одного туза в выборке из 4-х карт может получиться либо в случае выбора 1, 2, 3 или 4-х. Согласно формуле для вычисления количества сочетаний:

$$p = \frac{C_4^1*C_{48}^3+C_4^2*C_{48}^2+C_4^3*C_{48}^1+C_4^4*C_{48}^0}{C_{52}^4}$$

В этой формуле не учтен еще один возможноый случай, когда среди 4-х выбранных карт нет ни одного туза. Но если бы мы его учли, то вероятность была бы равна 1. 

$$1 = \frac{C_4^1*C_{48}^3+C_4^2*C_{48}^2+C_4^3*C_{48}^1+C_4^4*C_{48}^0+ C_4^0*C_{48}^{48}}{C_{52}^4}$$

Поэтому

$$p = \frac{C_4^1*C_{48}^3+C_4^2*C_{48}^2+C_4^3*C_{48}^1+C_4^4*C_{48}^0}{C_{52}^4}=1-\frac{C_4^0*C_{48}^4}{C_{52}^4}=1-\frac{(\frac{4!}{0!*4!})*(\frac{48!}{44!*4!})}{(\frac{52!}{48!*4!})}=1-\frac{45*46*47*48}{49*50*51*52}=0.281263 \approx 28.13 \%$$

In [5]:
import numpy as np

def combinations(n, k):
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

In [7]:
print(1 - combinations(4, 0)*combinations(48, 4)/combinations(52, 4))

print((combinations(4, 1)*combinations(48, 3)+combinations(4, 2)*combinations(48, 2)
+combinations(4, 3)*combinations(48, 1)+combinations(4, 4)*combinations(48, 0))/combinations(52, 4))

0.2812632745405854
0.2812632745405855


Но в задаче требуется накйти количество этих вариантов.

$$Количество = C_{52}^4-C_4^0*C_{48}^4$$

In [8]:
print(combinations(52, 4) - combinations(4, 0)*combinations(48, 4))

76145


Ответ: 76145

   
---


**Задача 2,3.**

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Случайным образом студент вытягивает 3 вопроса. 2) Какова вероятность, что все выбранные вопросы знакомы студенту? 3) Какова вероятность что два из трёх вопросов знакомы студенту?

Решение: Согласно формуле для вычисления количества сочетаний:

$$ p_{2} = \frac{C_{50}^3*C_{10}^{0}}{C_{60}^{3}}=\frac{(\frac{50!}{47!*3!})*(\frac{10!}{0!*10!})}{(\frac{60!}{57!*3!})}=\frac{6*50*49*48}{60*59*58*6}=\frac{19600}{34220}=0.57276446 \approx57 \%$$

$$ p_{3} = \frac{C_{50}^2*C_{10}^{1}}{C_{60}^{3}}=\frac{(\frac{50!}{48!*2!})*(\frac{10!}{1!*9!})}{(\frac{60!}{57!*3!})}=\frac{6*50*49*10}{60*59*58*2}=\frac{12250}{34220}=0.357977790 \approx 36\%$$

In [9]:
print(f'2) {(combinations(50, 3)*combinations(10, 0))/combinations(60, 3)}')
print(f'3) {(combinations(50, 2)*combinations(10, 1))/combinations(60, 3)}')

2) 0.5727644652250146
3) 0.35797779076563413


Ответ: 
задача 2) 0.57276446≈57%;
задача 3) 0.357977790≈36%.

---

**Задача 4.**

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

Формула Байеса:

$$P(B\:|\:A) = \frac{P(A\:|\:B) \cdot P(B)}{P(A)}$$

Решение:

Событие A - положительный результат теста.
Событие $В_1$ - мы здоровы.
Событие $В_2$ - мы действительно болеем.

$$P(B_1) = 0,999$$
$$P(A|B_1) = 0,01$$
$$P(B_2) = 0,001$$
$$P(A|B_2) = 0,99$$

$$P(A) = P(B_1) \cdot P(A | B_1) + P(B_2) \cdot P(A | B_2)$$

$$P(B_2\:|\:A) = \frac{P(A\:|\:B_2) \cdot P(B_2)}{P(A)}$$

$$P(B_2\:|\:A) = \frac{0,99 \cdot 0,001}{0,999*0,01+0,001*0,99} = 0.0901639344 \approx 9\%$$

In [10]:
print(f'P(B2|A) = {0.99*0.001/(0.999*0.01+0.001*0.99)}')

P(B2|A) = 0.09016393442622951


Ответ: 0.0901639344≈9%